pyproject.toml

```diff
@@ -1,5 +1,5 @@
 [build-system]
-requires = ["setuptools>=75.8.1"]
+requires = ["setuptools>=77.0.3"]
```

Comment:
This will mean that an editable install with `-e` will not work on Python 3.8 on the 4.2 branch, but PyPI is forcing our hand; we need to be able to upload to PyPI after February 2026, and I don't think we should switch build tools altogether.

Perhaps we can wait to backport this to 4.2 until we truly need it, e.g if there are further releases of 4.2.x in the six weeks after the PyPI deadline of 2026-Feb-18 before 4.2 goes EOL.